core/src/main/java/org/jruby/RubyGlobal.java

```diff
@@ -434,8 +435,9 @@ public IRubyObject to_s() {
         }
 
         @JRubyMethod
-        public RubyHash to_h(){
-            return to_hash();
+        public RubyHash to_h(ThreadContext context, Block block){
```

Comment:
we should consider keeping these public method (on RubyXXX types) signatures around (marked as deprecated)
due exts occasionally calling methods directly e.g. 
`@Deprecated public RubyHash to_h() { return to_h(getCurrentContext(), NULL_BLOCK) }`

Reply:
Thanks @kares, I didn't think about that, good catch. I just updated only this method following your recommendation, but just to clarify: do you consider I should to the same backward compatible change for all the public ```to_h``` modified methods in this PR?

Reply:
yeah, that would be the same - to usually deprecate public methods on (core) RubyXXX.java types

Reply:
@kares This should be fixed in the most recent version of this PR.